src/query/storages/fuse/src/io/read/block/block_reader_merge_io_async.rs

```diff
@@ -112,7 +113,11 @@ impl BlockReader {
                     .owner_memory
                     .get_chunk(*chunk_idx, &merge_io_result.block_path)?;
                 let data = chunk_data.slice(range.clone());
-                column_data_cache.insert(column_cache_key.as_ref().to_owned(), data);
+
+                column_data_cache.insert(
+                    column_cache_key.as_ref().to_owned(),
+                    ColumnData::from_merge_io_read_result(data),
```

Comment:
`chunk_data` is read from merge io, it's wasteful to store all chunk_data into memory, but it's ok to store all chunk_data into disk by ranges.  Maybe later we can optimize it.